odoo/addons/base/models/res_users.py

```diff
@@ -1272,6 +1284,50 @@ def _update_user_groups_view(self):
             new_context['lang'] = None
             view.with_context(new_context).write({'arch': xml_content})
 
+    @api.model
+    def _update_user_groups_tree_view(self):
```

Comment:
What is the purpose of this code ? not sure to find relevant specs in task ..

Reply:
Purpose here is to add a "portal users" filter on the users tree, since this field is created programmatically at run time, we cannot use xpath for this, we also need to add this field programmatically.

Reply:
Why not using share field ?

Reply:
From my understanding of the specs, we want to target portal users specifically, the share field would not work in this case as it also includes public users.